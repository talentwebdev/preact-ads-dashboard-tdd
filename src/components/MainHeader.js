import React from 'react'
import '../styles/MainHeader.css';
import BannerImg from '../assets/banner3.jpg';
import BannerImg2 from '../assets/banner5.jpg';
import SubLogo from '../assets/logo6.png';
import {Grid} from '@material-ui/core';
function MainHeader() {
    return (
        <div className="mainHeader">
            <div className="topSide" style={{background: `url(${BannerImg})`, height: '15vh'}}>
                <h1>BHHS The Preferred Realty Dashboard</h1>
            </div>
            <div className="bottomSide">
            <Grid item lg={12} container spacing={2}>
                <Grid item lg={3} md={4} sm={5} xs={12}>
                    <div className="btLeftSide">
                        <div className="subLeft">
                            <div className="subLeftTop">
                                <h5>BERKSHIRE</h5>
                                <h5>HATHAWAY</h5>
                            </div>
                            <div className="subLeftBottom">
                                <p>HomeServices</p>
                            </div>
                        </div>
                        <div className="subRight">
                            <p>The preferred<br />Realty</p>
                        </div>
                    </div>
                </Grid>
                <Grid item lg={9} md={8} sm={7} xs={12}>
                    <div className="btRightSide" style={{background: `url(${BannerImg2})`}} >
                        <div id="leftSide">
                            <img src={SubLogo} alt="" />
                        </div>
                        <div id="rightSide">
                            <h2>Expend Your Reach With Boost Agent Recruitment Campains</h2>
                        </div>
                    </div>
                </Grid>
            </Grid>
            </div>
        </div>
    )
}

export default MainHeader;
