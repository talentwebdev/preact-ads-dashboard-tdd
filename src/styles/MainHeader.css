.mainHeader{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .topSide{width:80%;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .topSide h1{color:#fff;font-size:55px}.mainHeader .bottomSide{width:70%;height:auto;margin-top:2vh;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .bottomSide .btLeftSide{background-color:#fff;height:15vh;width:100%;border:2px solid blue;border-radius:5px;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .bottomSide .btLeftSide .subLeft{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop{border-right:1px solid #4D1936}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:23px;padding-right:10px;font-weight:500;font-family:'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;color:#4d1936}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:13px;color:#4d1936;font-weight:600}.mainHeader .bottomSide .btLeftSide .subLeft h5,.mainHeader .bottomSide .btLeftSide .subLeft p{margin:0;padding:0}.mainHeader .bottomSide .btLeftSide .subRight{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:start;-ms-flex-align:start;align-items:flex-start}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:18px;color:#4d1936;padding-left:10px;margin-bottom:15px}.mainHeader .bottomSide .btRightSide{background-color:#fff;height:15vh;width:100%;border:2px solid blue;border-radius:5px;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .bottomSide .btRightSide #leftSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-flex:20%;-ms-flex:20%;flex:20%}.mainHeader .bottomSide .btRightSide #leftSide img{margin-left:20px;width:10vw}.mainHeader .bottomSide .btRightSide #rightSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-flex:80%;-ms-flex:80%;flex:80%;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.mainHeader .bottomSide .btRightSide #rightSide h2{color:#fff;font-size:2.3vw;font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif}@media only screen and (min-width: 1150px){.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:20px}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:12px}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:15px}}@media only screen and (max-width: 1150px){.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:20px}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:12px}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:15px}.mainHeader .topSide h1{font-size:50px}.mainHeader .bottomSide .btLeftSide{height:14vh}.mainHeader .bottomSide .btRightSide{height:14vh}}@media only screen and (max-width: 950px){.mainHeader .bottomSide .btRightSide #rightSide h2{color:#fff;font-size:2vw;font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif}.mainHeader .bottomSide .btLeftSide{height:13.5vh}.mainHeader .bottomSide .btRightSide{height:13.5vh}}@media only screen and (max-width: 800px){.mainHeader .bottomSide .btLeftSide .subRight p{font-size:14px}}@media only screen and (max-width: 750px){.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:18px}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:10px}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:12px}.mainHeader .topSide h1{font-size:40px}.mainHeader .bottomSide .btLeftSide{height:13vh}.mainHeader .bottomSide .btRightSide{height:13vh}}@media only screen and (max-width: 650px){.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:15px}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:9px}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:11px;padding:5px}.mainHeader .topSide h1{font-size:38px}.mainHeader .bottomSide .btLeftSide{height:12vh}.mainHeader .bottomSide .btRightSide{height:12vh}}@media screen and (max-width: 600px){.mainHeader .topSide{width:100%}.mainHeader .topSide h1{font-size:7vw}.mainHeader .bottomSide{width:100%}.mainHeader .bottomSide .btLeftSide{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;width:60%;margin:auto;height:17vw}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5{font-size:4vw}.mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p{font-size:2.5vw}.mainHeader .bottomSide .btLeftSide .subRight p{font-size:3vw;padding-left:5vw}.mainHeader .bottomSide .btRightSide{width:95%;margin:auto;height:17vw}.mainHeader .bottomSide .btRightSide #leftSide img{margin-left:15px;width:17vw}.mainHeader .bottomSide .btRightSide #rightSide h2{font-size:4.5vw}}@media only screen and (min-width: 600px){.mainHeader .bottomSide .btLeftSide{overflow:auto}.mainHeader .bottomSide .btRightSide{overflow:hidden}}
