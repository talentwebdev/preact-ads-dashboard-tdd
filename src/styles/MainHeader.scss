.mainHeader {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    .topSide {
        width: 80%;
        display: flex;
        justify-content: center;
        align-items: center;
        h1 {
            color: #fff;
            font-size: 55px;
        }
    }
    .bottomSide {
        width: 70%;
        height: auto;
        margin-top: 2vh;
        display: flex;
        justify-content: center;
        align-items: center;
        .btLeftSide {
            background-color: #fff;
            height: 15vh;
            width: 100%;
            border: 2px solid blue;
            border-radius: 5px;
            display: flex;
            flex-direction: row;
            justify-content: center;
            align-items: center;
            .subLeft {
                display: flex;
                flex-direction: column;
                justify-content: center;
                align-items: center;
                .subLeftTop {
                    border-right: 1px solid #4D1936;
                    h5 {
                        font-size: 23px;
                        padding-right: 10px;
                        font-weight: 500;
                        font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
                        color: #4d1936;
                    }
                }
                .subLeftBottom p {
                    font-size: 13px;
                    color: #4d1936;
                    font-weight: 600;
                }
                h5,
                p {
                    margin: 0;
                    padding: 0;
                }
            }
            .subRight {
                display: flex;
                justify-content: center;
                align-items: flex-start;
                p {
                    font-size: 18px;
                    color: #4d1936;
                    padding-left: 10px;
                    margin-bottom: 15px;
                }
            }
        }
        .btRightSide {
            background-color: #fff;
            height: 15vh;
            width: 100%;
            border: 2px solid blue;
            border-radius: 5px;
            display: flex;
            flex-direction: row;
            justify-content: center;
            align-items: center;
            #leftSide {
                display: flex;
                flex: 20%;
                img {
                    margin-left: 20px;
                    width: 10vw;
                }
            }
            #rightSide {
                display: flex;
                flex: 80%;
                justify-content: center;
                align-items: center;
                h2 {
                    color: #fff;
                    font-size: 2.3vw;
                    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
                }
            }
        }
    }
}

@media only screen and (min-width: 1150px) {
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5 {
        font-size: 20px;
    }
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p {
        font-size: 12px;
    }
    .mainHeader .bottomSide .btLeftSide .subRight p {
        font-size: 15px;
    }
}

@media only screen and (max-width: 1150px) {
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5 {
        font-size: 20px;
    }
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p {
        font-size: 12px;
    }
    .mainHeader .bottomSide .btLeftSide .subRight p {
        font-size: 15px;
    }
    .mainHeader .topSide h1 {
        font-size: 50px;
    }
    .mainHeader .bottomSide .btLeftSide {
        height: 14vh;
    }
    .mainHeader .bottomSide .btRightSide {
        height: 14vh;
    }
}

@media only screen and (max-width: 950px) {
    .mainHeader .bottomSide .btRightSide #rightSide h2 {
        color: #fff;
        font-size: 2vw;
        font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    }
    .mainHeader .bottomSide .btLeftSide {
        height: 13.5vh;
    }
    .mainHeader .bottomSide .btRightSide {
        height: 13.5vh;
    }
}

@media only screen and (max-width: 800px) {
    .mainHeader .bottomSide .btLeftSide .subRight p {
        font-size: 14px;
    }
}

@media only screen and (max-width: 750px) {
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5 {
        font-size: 18px;
    }
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p {
        font-size: 10px;
    }
    .mainHeader .bottomSide .btLeftSide .subRight p {
        font-size: 12px;
    }
    .mainHeader .topSide h1 {
        font-size: 40px;
    }
    .mainHeader .bottomSide .btLeftSide {
        height: 13vh;
    }
    .mainHeader .bottomSide .btRightSide {
        height: 13vh;
    }
}

@media only screen and (max-width: 650px) {
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftTop h5 {
        font-size: 15px;
    }
    .mainHeader .bottomSide .btLeftSide .subLeft .subLeftBottom p {
        font-size: 9px;
    }
    .mainHeader .bottomSide .btLeftSide .subRight p {
        font-size: 11px;
        padding: 5px;
    }
    .mainHeader .topSide h1 {
        font-size: 38px;
    }
    .mainHeader .bottomSide .btLeftSide {
        height: 12vh;
    }
    .mainHeader .bottomSide .btRightSide {
        height: 12vh;
    }
}

@media screen and (max-width: 600px) {
    .mainHeader {
        .topSide {
            width: 100%;
            h1 {
                font-size: 7vw;
            }
        }
        .bottomSide {
            width: 100%;
            .btLeftSide {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 60%;
                margin: auto;
                height: 17vw;
                .subLeft {
                    .subLeftTop h5 {
                        font-size: 4vw;
                    }
                    .subLeftBottom p {
                        font-size: 2.5vw;
                    }
                }
                .subRight p {
                    font-size: 3vw;
                    padding-left: 5vw;
                }
            }
            .btRightSide {
                width: 95%;
                margin: auto;
                height: 17vw;
                #leftSide img {
                    margin-left: 15px;
                    width: 17vw;
                }
                #rightSide h2 {
                    font-size: 4.5vw;
                }
            }
        }
    }
}

@media only screen and (min-width: 600px) {
    .mainHeader .bottomSide .btLeftSide {
        overflow: auto;
    }
    .mainHeader .bottomSide .btRightSide {
        overflow: hidden;
    }
}