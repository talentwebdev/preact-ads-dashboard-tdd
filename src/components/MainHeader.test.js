import { shallow } from "enzyme";
import MainHeader from "./MainHeader";

describe("MainHeader", () => {
  let mainHeaderContainer;

  beforeEach(() => (mainHeaderContainer = shallow(<MainHeader />)));

  it("should render the div element", () => {
    expect(mainHeaderContainer.find("div").length).toBeGreaterThanOrEqual(1);
  });
  it("renders MainHeader correctly", () => {
    expect(mainHeaderContainer).toMatchSnapshot();
  });
});
