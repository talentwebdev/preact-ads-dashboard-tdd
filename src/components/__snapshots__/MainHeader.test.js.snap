// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader renders Header correctly 1`] = `ShallowWrapper {}`;

exports[`MainHeader renders MainHeader correctly 1`] = `ShallowWrapper {}`;
